### PR TITLE
public unpair AD

### PR DESCRIPTION
**Pull request by LorranSutter**

### CODE REVIEW

File: contracts/implementations/nodes/AftermarketDevice.sol [hunk 1]
```diff
@@ -404,6 +404,53 @@ contract AftermarketDevice is
...
+        if (!INFT(adIdProxyAddress).exists(aftermarketDeviceNode))
+            revert Errors.InvalidNode(adIdProxyAddress, aftermarketDeviceNode);
+        if (!INFT(vehicleIdProxyAddress).exists(vehicleNode))
+            revert Errors.InvalidNode(vehicleIdProxyAddress, vehicleNode);
+        if (
+            ms.links[vehicleIdProxyAddress][vehicleNode] !=
+            aftermarketDeviceNode
+        ) revert VehicleNotPaired(vehicleNode);
+        if (ms.links[adIdProxyAddress][aftermarketDeviceNode] != vehicleNode)
+            revert AdNotPaired(aftermarketDeviceNode);
```
Comment (elffjs) [inline contracts/implementations/nodes/AftermarketDevice.sol:435]: If we pass the first check but fail the second, that's catastrophic, no?

**Review by elffjs — APPROVED** [review]

File: contracts/implementations/nodes/AftermarketDevice.sol (hunk 1 above)
Comment (LorranSutter) [reply contracts/implementations/nodes/AftermarketDevice.sol:435]: Kind of, bc those two links are always supposed to be updated together. Might be worth combine these two `ifs` then

Comment (elffjs) [reply contracts/implementations/nodes/AftermarketDevice.sol:435]: Well, you have the two events, let's use them. I just wanted to point it out.

**Review by Allyson-English — APPROVED** [review]

**Comment by elffjs** [comment]
Any objections to merging?

**Pull request merged**